pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -780,9 +779,8 @@ func (cb *ClusterBuilder) applyTrafficPolicy(opts buildClusterOpts) {
 	if connectionPool == nil {
 		connectionPool = &networking.ConnectionPoolSettings{}
 	}
-	cb.applyConnectionPool(opts.mesh, opts.mutable, connectionPool)
+	cb.applyConnectionPool(opts.mesh, opts.mutable, connectionPool, opts.port)
 	if opts.direction != model.TrafficDirectionInbound {
-		cb.applyH2Upgrade(opts, connectionPool)
```

Comment:
moved applyH2Upgrade into applyConnectionPool as h2UpgradePolicy field is under connectionPool